src/OSPSuite.Presentation/Binders/ObservedDataDragDropBinder.cs

```diff
@@ -24,6 +25,25 @@ public IReadOnlyList<DataRepository> DroppedObservedDataFrom(IDragEvent e)
          return getObservedDataNodesFrom(e).Select(observedDataNode => observedDataNode.Tag.Subject).ToList();
       }
 
+      public Cache<string, List<DataRepository>> DroppedObservedDataWithFolderPathFrom(IDragEvent e)
+      {
+         var dataNodes = e.Data<IEnumerable<ITreeNode>>();
+
+         if (dataNodes == null)
+            return new Cache<string, List<DataRepository>>();
+
+         var treeNodes = dataNodes as IList<ITreeNode> ?? dataNodes.ToList();
+
+         if (areAllObservedDataNodes(treeNodes))
+            return observedDataNodesWithFolderPathFromObservedDataNodes(treeNodes);
+
+         if (areAllObservedDataClassificationNodes(treeNodes) || areAllRootNodes(treeNodes))
```

Comment:
Very nice and easy to read

Reply:
Hahahaha, true true. Renamed it to, observedDataNodesFromNodes().